### ISSUE TITLE
can not define loss by self 

### ISSUE THREAD

**Issue by haoxurt** [zhanghang1989/PyTorch-Encoding#350]
Thanks for share. When I try add some loss such as focal loss, I encounter the following error.Even if I realize CrossEntropy by self, the error is existed. Hoping your help!
error 
`/pytorch/aten/src/THCUNN/SpatialClassNLLCriterion.cu:104: void cunn_SpatialClassNLLCriterion_updateOutput_kernel(T *, T *, T *, long *, T *, int, int, int, int, int, long) [with T = float, AccumT = float]: block: [3,0,0], thread: [160,0,0] Assertion t >= 0 && t < n_classes failed.`

code :
`class SegmentationLosses(nn.CrossEntropyLoss):
    """2D Cross Entropy Loss with Auxilary Loss"""
    def __init__(self, se_loss=False, se_weight=0.2, nclass=-1,
                 aux=False, aux_weight=0.4, weight=None,
                 ignore_index=-1):
        super(SegmentationLosses, self).__init__(weight, None, ignore_index)
        #super(SegmentationLosses, self).__init__()
        self.se_loss = se_loss
        self.aux = aux
        self.nclass = nclass
        self.se_weight = se_weight
        self.aux_weight = aux_weight
        self.bceloss = nn.BCELoss(weight) 
        self.ohemloss = OhemCELoss(ratio=0.5)
        self.focalloss = SoftmaxFocalLoss()
        self.cross = nn.CrossEntropyLoss()

    def forward(self, *inputs):
        if not self.se_loss and not self.aux:
            return super(SegmentationLosses, self).forward(*inputs)
        elif not self.se_loss:
            pred1, pred2, target = tuple(inputs)
            # loss1 = super(SegmentationLosses, self).forward(pred1, target)
            # loss2 = super(SegmentationLosses, self).forward(pred2, target)
            loss1 = self.cross(pred1, target)
            loss2 = self.cross(pred2, target)
            # loss1 = self.focalloss(pred1, target)
            # loss2 = self.focalloss(pred2, target)
            return loss1 + self.aux_weight * loss2`

`lass SoftmaxFocalLoss(nn.Module):
    def __init__(self, gamma=2):
        super(SoftmaxFocalLoss, self).__init__()
        self.gamma = gamma
        self.nll = nn.NLLLoss()

    def forward(self, logits, labels):
        scores = F.softmax(logits, dim=1)
        factor = torch.pow(1. - scores, self.gamma)
        log_score = F.log_softmax(logits, dim=1)
        log_score = factor * log_score
        loss = self.nll(log_score, labels)
        return loss
`

**Comment by zhanghang1989** [comment]
That error is because the ground truth exceeds the limit (number of classes / number of output channels).

Good luck for debuging.

**Comment by haoxurt** [comment]
I do not change ground truth(target)，it come from 'tuple(inputs)'.
